cmd/link.go

```diff
@@ -14,22 +16,21 @@ var (
 )
 
 var linkCmd = &cobra.Command{
-	Use:   "link <target component> --component [source component]",
-	Short: "Link target component to source component",
-	Long: `Link target component to source component
+	Use:   "link <service> --component [component]",
```

Comment:
just a thought in my mind:

rather than, `odo link <service> --component <component>`, `odo service link <service> --component <component>`
would be much cleaner.
But considering current situation, since all `odo <verb>` deal with components.
I think it should be,

```
odo link <component> --service <service>
```

and also I think one component can consume mulitple services, so user can do
```
odo link component --service service1
odo link component --service service2
```

WDYT @geoand ?

Reply:
Hi @surajnarwade,

I am not opposed to the suggestion, it makes perfect sense. For the moment however, it as decided [here](https://github.com/redhat-developer/odo/issues/652#issuecomment-429813904), move along with this way, which is why I implemented this way.
If you were to raise your concern on that issue and it's decided to use your suggestion, I will gladly update this PR

Reply:
@surajnarwade globally speaking, I think all commands should be reviewed for coherence and to avoid postional parameters